systemtest/loadgen/eventhandler/handler.go

```diff
@@ -71,8 +71,10 @@ type Handler struct {
 	config     Config
 	writerPool sync.Pool
 
-	batches      []batch
-	minTimestamp time.Time // across all batches
+	batches       []batch
+	queuedBatches []batch
+	queuedEvents  int
```

Comment:
Sorry I didn't notice this earlier: Handler may be used by multiple concurrent callers, so we can't store queuedBatches and queuedEvents on the struct. I think they would have to be stored on the stack in SendBatches.